pkg/codegen/go/gen_program_read_dir.go

```diff
@@ -0,0 +1,70 @@
+package gen
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/pulumi/pulumi/pkg/v2/codegen/hcl2/model"
+	"github.com/pulumi/pulumi/pkg/v2/codegen/hcl2/syntax"
+)
+
+type readDirTemp struct {
+	Name  string
+	Value *model.FunctionCallExpression
+}
```

Comment:
We should think about unifying these temp kinds in the future.

Reply:
With what goal? Eliminating redundancy in the spillers?

Reply:
Yes, that's right.